feat: add C implementation for `stats/base/dists/triangular/logpdf`

0PrashantYadav0 opened the pull request:
Resolves #3815

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/triangular/logpdf along with relevant tests, benchmarks and examples.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3815

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C implementation for triangular logpdf

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/package.json:
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@stdlib/stats/base/dists/triangular/logpdf",
+  "name": "@stdlib/stats/base/dists/triangular/pdf",
```
These changes need to be reverted.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/README.md:
````diff
@@ -158,6 +158,109 @@ for ( i = 0; i < 25; i++ ) {
+### Usage
+
+```c
+#include "stdlib/stats/base/dists/geometric/logpdf.h"
````
Copy & paste error: `geometric` should be `triangular` here and below.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/include/stdlib/stats/base/dists/triangular/logpdf.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Evaluates the natural logarithm of the probability density function (PDF) for the triangular distribution with parameters a, b, and c.
```
```suggestion
* Evaluates the natural logarithm of the probability density function (PDF) for a triangular distribution with lower limit `a` and upper limit `b` and mode `c` at a value `x`.
```

Planeshifter reviewed (changes requested):
Some copy & paste errors need to be resolved, afterwards PR should be ready to land.

0PrashantYadav0 committed:
chore: made changes as per code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@Planeshifter I make all the changes you asked for please review and merge. Thank you

0PrashantYadav0 commented:
@Planeshifter make the changes. Please review and merge this pr. Thank you.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/README.md:
```diff
@@ -158,6 +158,109 @@ for ( i = 0; i < 25; i++ ) {
+The function accepts the following arguments:
+
+-   **x**: `[in] double` input value.
+-   **a**: `[in] double` lower limit
+-   **b**: `[in] double` upper limit
+-   **c**: `[in] double` mode
```
```suggestion
-   **a**: `[in] double` lower limit.
-   **b**: `[in] double` upper limit.
-   **c**: `[in] double` mode.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/README.md:
```diff
@@ -158,6 +158,109 @@ for ( i = 0; i < 25; i++ ) {
+    for ( i = 0; i < 25; i++ ) {
+        x = random_uniform( 0.0, 30.0 );
+        a = random_uniform( 0.0, 10.0 );
+        b = random_uniform( a, 40.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
```
```suggestion
        b = random_uniform( a + STDLIB_CONSTANT_FLOAT64_EPS, 40.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,145 @@
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( 0.0, 30.0 );
+		a[ i ] = random_uniform( 0.0, 10.0 );
+		b[ i ] = random_uniform( a[ i ], 40.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
```
```suggestion
		b[ i ] = random_uniform( a[ i ] + STDLIB_CONSTANT_FLOAT64_EPS, 40.0 );
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Let's get it in...
